java/lang/path_traversal.yml

```diff
@@ -1,34 +1,56 @@
 patterns:
   - pattern: |
-      new File($<...>$<FILE_USER_INPUT>$<...>);
+      new $<METHOD>($<...>$<FILE_USER_INPUT>$<...>);
     filters:
+      - variable: METHOD
+        regex: (java.)?(io.)?(File|FileReader|FileWriter|FileInputStream|FileOutputStream)
       - variable: FILE_USER_INPUT
         detection: java_lang_path_traversal_user_input
         scope: result
   - pattern: |
-      new FileOutputStream($<...>$<FILE_OUTPUT_USER_INPUT>$<...>)
+      new $<METHOD>($<...>$<FILE_USER_INPUT>$<...>);
```

Comment:
Should this have `new`? (and following patterns)

Reply:
Good shout - thanks!